deploy/control.Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21
+FROM docker.io/golang:1.22
```

Comment:
If we're going to upgrade the Go version (which I don't think is a change small enough to tag as an extra to this pr), we'll need to update our ci pipeline to also build on go 1.22

Reply:
gah, i’ll fix this when i’m at a keyboard.

i’d like to do the go version bump separately, as it’s orthogonal, and i also want the new loop capture semantics

Reply:
fixed!